src/controllers/expensesService.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const userService = require('./userService.js');
+
+let expenses = [];
+
+const init = () => {
+  expenses = [];
+};
+
+const createExpence = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!title || !userService.exist(userId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const createdExpense = {
+    id: Math.floor(Math.random() * 100),
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(createdExpense);
+  res.statusCode = 201;
+  res.send(createdExpense);
+};
+
+const getAllExpensesByParams = (req, res) => {
+  const {
+    userId,
+    category,
+    from,
+    to,
+  } = req.query;
+
+  const numberId = Number(userId);
+
+  if (!expenses.length) {
+    res.statusCode = 200;
+    res.send([]);
+
+    return;
+  }
+
+  res.statusCode = 200;
+
+  if (userService.findUser(numberId)) {
+    let userExpenses = expenses.filter(expense => expense.userId === numberId);
+
+    if (category) {
+      userExpenses = userExpenses.filter(
+        expense => expense.category === category);
+    }
+    res.send(userExpenses);
+
+    return;
+  }
+
+  if (from && to) {
+    const foundExpences = expenses.filter(
+      expense => expense.spentAt >= from && expense.spentAt <= to);
+
+    res.send(foundExpences);
+
+    return;
+  }
+
+  res.send(expenses);
+};
+
+const getExpenceById = (req, res) => {
+  const { id } = req.params;
+  const numberId = Number(id);
+
+  const foundExpense = expenses.find(expense => expense.id === numberId);
+
```

Comment:
you use this logic several times in your code so it would be better to make a helper function